frontend-project/src/models/users.ts

```diff
@@ -0,0 +1,46 @@
+import { fetchAll } from '@/services/users';
+import { Effect, EffectsCommandMap } from 'dva';
+import { Reducer, AnyAction } from 'redux';
+
+export interface User {
+  username: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  id: number;
+}
+
+export interface UsersState {
+  users: User[];
+}
+
+export interface UsersModelType {
+  namespace: string;
+  state: UsersState;
+  effects: {
+    fetchAll: Effect;
+  };
+  reducers: {
+    saveAll: Reducer<UsersState>;
+  };
+}
+
+const UsersModel = {
```

Comment:
const UserModel: UsersModelType powinno być

Reply:
typ dla saveAll krzyczy, ze się nie zgadza.

`export interface UsersModelType {
  namespace: string;
  state: UsersState | [];
  effects: {
    fetchAll: Effect;
  };
  reducers: {
    saveAll: Reducer<UsersState>;
  };
}

const UsersModel: UsersModelType = {
  namespace: 'users',
  state: [],
  effects: {
    *fetchAll(_: AnyAction, { call, put }: EffectsCommandMap) {
      const response = yield call(fetchAll);
      yield put({
        type: 'saveAll',
        payload: response,
      });
    },
  },
  reducers: {
    saveAll(_: AnyAction, { payload }: EffectsCommandMap) {
      return payload.results;
    },
  },
};`

Reply:
Mogę zmienić typ na any, ale nie wiem, czy jest to rozsądne, masz jakąś sugestie? Sam też będę szukał.